### ISSUE TITLE
Adding a catalog result in error: [path/to/folder] is not readable or does not exist

### ISSUE THREAD

**Issue by tzeus** [ampache/ampache#1645]
# Environment

 * Ampache version:  **Ampache 3.8.5-develop**
 * Web server + version: 

**sudo dpkg -l | grep apache**

`ii  apache2                               2.4.25-3+deb9u3                      armhf        Apache HTTP Server
`
**sudo dpkg -l | grep php**

`ii  php7.0                                7.0.19-1                             all          server-side, HTML-embedded scripting language (metapackage)`

 **sudo dpkg -l | grep mysql**

`ii  mysql-server                          5.5.9999+default                     armhf        MySQL database server binaries and system database setup [transitional]`

 * Server operating system:

**uname -a**

`Linux music 4.9.59-v7+ #1047 SMP Sun Oct 29 12:19:23 GMT 2017 armv7l GNU/Linux
`

**lsb_release -irc**

```
Distributor ID: Raspbian
Release:        9.1
Codename:       stretch
```

 * Client operating system:
Windows 10. Firefox Quantum, Google Chrome

 * Client type:
   - Web

# Settings

**music folder permissions from hdd:**

`drwxrwxrwx 1 pi pi      131072 Jun  9  2017 Music
`

Config is default, excepting username and password for db
# Description

Cannot add a new local catalog. Trying to add a new catalog just result in the following error
**Error: /media/pi/Music/Music is not readable or does not exist**

Both the ampache folder and the music folder are owned by root.
I also tried with 777 persmission on music folder, to the same effect.
I tried mounting normally, or mounting with fstab, same effects.

# Logs

2017-12-14 19:43:10 [admin] (Core) -> Verified SID 91d429075de91360dcd81179e1634869 for post form add_catalog
```
2017-12-14 19:43:10 [admin] (catalog) -> Cannot add catalog at unopenable path /media/pi/Music/Music
2017-12-14 19:43:10 [admin] (autoload) -> 'Beets\Catalog' not found!
2017-12-14 19:43:10 [admin] (Core) -> Registered post form add_catalog with SID 2f568d1b9d8fd1195503120817372b89 and expiration 1513276990 (3600 seconds from now)
2017-12-14 19:43:12 [admin] (session) -> bpbr4jd4rtoksqs02ek0cliq51 has been extended to Thu, 14 Dec 2017 20:43:12 +0200 extension length 3600
2017-12-14 19:43:12 [admin] (ajax.server.php) -> Called for page: {index}
2017-12-14 19:43:35 [admin] (session) -> bpbr4jd4rtoksqs02ek0cliq51 has been extended to Thu, 14 Dec 2017 20:43:35 +0200 extension length 3600
2017-12-14 19:43:35 [admin] (ajax.server.php) -> Called for page: {index}
2017-12-14 19:43:44 [admin] (session) -> 1imaag34rui2v12a6gaidam7r6 has been extended to Thu, 14 Dec 2017 20:43:44 +0200 extension length 3600
2017-12-14 19:43:44 [admin] (ajax.server.php) -> Called for page: {index}
2017-12-14 19:43:57 [admin] (session) -> bpbr4jd4rtoksqs02ek0cliq51 has been extended to Thu, 14 Dec 2017 20:43:57 +0200 extension length 3600
2017-12-14 19:43:57 [admin] (session) -> Session created: f8a3f9163bacb9f2a47be4dafd464d66
2017-12-14 19:43:57 [admin] (Core) -> Verified SID 2f568d1b9d8fd1195503120817372b89 for post form add_catalog
2017-12-14 19:43:57 [admin] (catalog) -> Cannot add catalog at unopenable path /media/pi/Music/Music
2017-12-14 19:43:57 [admin] (autoload) -> 'Beets\Catalog' not found!
2017-12-14 19:43:57 [admin] (Core) -> Registered post form add_catalog with SID f5bedc7d202dbbfe43aba567cd643008 and expiration 1513277037 (3600 seconds from now)
```
![capture1](https://user-images.githubusercontent.com/4553914/34006579-8b208156-e107-11e7-9643-d813d56dff2b.PNG)

![capture2](https://user-images.githubusercontent.com/4553914/34006592-9283cb88-e107-11e7-898c-7b1b1b1b083b.PNG)

**Comment by wagnered** [comment]
Could you try running the Apache server as user:group pi:pi?

Normally, for security reasons, the folders to be accessed are set with
only user:group access and no public access. The server is set to run
with the same user;group as the folder owner.

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

On 12/14/2017 10:28 AM, Tudorel Oprisan wrote:
......

**Comment by tzeus** [comment]
I gave www-data ownership to the music folder and the result is the same. I started the apache as pi:pi, and changed permission on the music folder to pi:pi. Still the same error in the log file and in the front-end.
The only thing that worked was to mount it via /etc/fstab with defaults.
That resulted in the Music folder mounting on root:root with 777 permissions.

Note that this seems to be a ntfs-3g / fuse issue, because I encountered a problem trying to chmod resulting in an **transport endpoint not connected** error. This lead me to believe that this is most likely an OS issue and not an ampache issue.

For the moment I managed to create a catalog containing all the music. 
So I think you could close this issue.

**Comment by wagnered** [comment]
@tzeus 
That's good to hear....

It's also a bit of information that wan't conveyed to me. 😃

Also,  I'm not sure why the HDD was formatted ntfs on a Linux system.  If you want to allow Windows to access it, It might be better to format the drive as ext4 and install samba for Windows access..

FYI: You can close an issue you opened and then re-open it if necessary.

**Comment by tzeus** [comment]
@wagnered  That information was unknown to me too :)
I posted again because that's when it became known to me. And I thought it's not a bug, at least not in the Ampache app, but in the ntfs drivers. Or maybe in the USB HDD, who knows.
It only happens when it tries to change the permission on the "System Volume Information".
It was formatted as NTFS, because it's the most convenient way to use data for me. 
The only reason I turned to Ampache is because the company where I work now, has some strict no usb device policy, so I have to use a remote streaming service that supports flac :).

Thanks for the support
Nice app btw, keep it going!
